### DISCUSSION TITLE
ungrouped /var/cache/private/fwupdmgr files

### DISCUSSION THREAD

**Discussion by dunter-d** [fwupd/fwupd#3098] [Q&A]
Searching for ungrouped files via `df --local -P | awk '{if (NR!=1) print $6}' | xargs -I '{}' find '{}' -xdev -nogroup` brings up serveral cached fwupdermgr files:

```
/var/cache/private/fwupdmgr
/var/cache/private/fwupdmgr/fwupd
/var/cache/private/fwupdmgr/fwupd/lvfs-metadata.xml.gz.asc
/var/cache/private/fwupdmgr/fwupd/lvfs-metadata.xml.gz
```
The permissions on these files are: `-rw-r--r-- 1 62803 62803`

Are these okay to delete and why are they ungrouped?

**Comment by superm1** [comment]
I believe these are likely created by this systemd unit: https://github.com/fwupd/fwupd/blob/master/data/motd/fwupd-refresh.service.in

The unit has `DynamicUser=yes` set, so a new user account is created and used every time the unit runs IIRC.

**Comment by auwsom** [comment]
Ungrouped files break CIS Benchmark testing: 6.1.12 Ensure no ungrouped files or directories exist

**Reply by superm1** [reply]
This is expected behavior from systemd units that use DynamicUser.  It's not a fwupd bug.

**Reply by evgenyz** [reply]
What is the point of having DynamicUser for a one-shot service and keeping the files around in the cache? Shouldn't these files be placed in /tmp in this case?

**Reply by hughsie** [reply]
It's not a cache really -- it's a download artifact. The way fwupd-refresh works is downloading the file from the internet and then sending it to fwupd over a socket -- if the fwupd version is up to date then no file is downloaded. In fwupd we could certainly delete the binary when invoked using fwupd-refresh -- but we couldn't delete the directories IIUC. I really think this should be cleaned up by fwupd -- like you say that there's no point keeping it if DynamicUser is being used.

**Comment by auwsom** [comment]
That may be true, but it still breaks CIS industry standard machine
hardening.

**Reply by superm1** [reply]
It should be a systemd bug then.

**Reply by hsreeramdevops** [reply]
Having same issue, do we know what happened?

**Comment by hsreeramdevops** [comment]
Having same issue, do we know what happened?

**Comment by vojtapolasek** [comment]
Systemd is a multitool, you are using one of its tools; dynamic users. It might be a valid use case, but it might be not and while reading this discussion, I don't see a reason for usage of dynamic users. As I understand it, if these files should be reused later by some other processes or fwupd instances, I don't think assigning ephemeral owners to them makes  sense.
